src/use-swr-pages.tsx

```diff
@@ -98,14 +98,27 @@ export function useSWRPages<OffsetType = any, Data = any, Error = any>(
 ): pagesResponseInterface<Data, Error> {
   const pageCountKey = `_swr_page_count_` + pageKey
   const pageOffsetKey = `_swr_page_offset_` + pageKey
+  const pageSWRsKey = `_swr_page_swrs_` + pageKey
 
   const [pageCount, setPageCount] = useState<number>(
     cache.get(pageCountKey) || 1
   )
   const [pageOffsets, setPageOffsets] = useState<OffsetType[]>(
     cache.get(pageOffsetKey) || [null]
   )
-  const [pageSWRs, setPageSWRs] = useState<responseInterface<Data, Error>[]>([])
+  const [pageSWRs, setPageSWRs] = useState<responseInterface<Data, Error>[]>(
+    cache.get(pageSWRsKey) || []
+  )
+
+  useEffect(() => {
+    setPageCount(cache.get(pageCountKey) || 1)
+  }, [pageCountKey])
+  useEffect(() => {
+    setPageOffsets(cache.get(pageOffsetKey) || [null])
+  }, [pageOffsetKey])
+  useEffect(() => {
+    setPageSWRs(cache.get(pageSWRsKey) || [])
+  }, [pageSWRsKey])
```

Comment:
Is there an advantage to creating multiple effect hooks, versus a single effect that resets all 3 states and depends on pageKey?

Also should this be a layoutEffect to prevent multiple updates when pageKey changes?